src/main/java/me/namuhuchutong/step3/RacingService.java

```diff
@@ -0,0 +1,32 @@
+package me.namuhuchutong.step3;
+
+import java.util.stream.IntStream;
+
+public class RacingService {
+
+    private final RacingRule racingRule;
+
+    public RacingService(RacingRule racingRule) {
+        this.racingRule = racingRule;
+    }
+
+    public RacingResult runRacing(UserInputInformation userInputInformation) {
+        validateTimes(userInputInformation.getTimes());
+        Cars initalizedCars = Cars.createCars(userInputInformation.getNumberOfCars());
+        return repeatRacing(initalizedCars, userInputInformation.getTimes());
+    }
+
+    private void validateTimes(int times) {
+        if (times <= 0) {
+            throw new IllegalArgumentException("자동차 경주 횟수는 음수이거나 0일 수 없습니다 - " + times);
+        }
+    }
+
+    private RacingResult repeatRacing(Cars initalizedCars, int times) {
+        RacingResult racingResult = new RacingResult();
+        IntStream.range(0, times)
+                 .mapToObj(i -> initalizedCars.raceAllCars(racingRule))
+                 .forEach(racingResult::addFinishedRacing);
+        return racingResult;
```

Comment:
```suggestion
    private RacingResult repeatRacing(Cars initalizedCars, int times) {
        List<Cars> collect = IntStream.range(0, times)
                .mapToObj(i -> initalizedCars.raceAllCars(racingRule))
                .collect(Collectors.toList());
        return new RacingResult(collect);
```

이렇게 변경해 볼 수 있을 것 같아요
이러면 RacingResult에 addFinishedRacing이라는 메서드도 제거할 수 있겠네요

여기에서도 IntStream을 사용하셨는데 Stream만으로도 생성이 가능해 보입니다.